### PR TITLE
Added PSP22 pallet-wrapper

### PR DESCRIPTION
**Pull request by PierreOssun**

### CODE REVIEW

**Comment by 123oli** [comment]
@PierreOssun I built and upload `PSP22 pallet-wrapper` contract but when `deposite` it throws `contracts.NoChainExtension` error. Why this is happening?
Screenshot 2022-12-01 at 12 55 53
Screenshot 2022-12-01 at 12 56 27
Screenshot 2022-12-01 at 12 57 30

**Comment by 123oli** [comment]
@PierreOssun I am using `@astar-network/swanky-cli/1.0.0-rc.4 darwin-x64 node-v18.12.1`, while init project I faced this error what could be the possible reason behind this issue?
Screenshot 2022-12-01 at 16 54 50

**Comment by PierreOssun** [comment]
@123oli use repo issue's for this https://github.com/AstarNetwork/swanky-cli/issues/47

File: contracts/psp22_pallet_wrapper/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ink_env = { version = "~3.4.0", default-features = false }
+ink_storage = { version = "~3.4.0", default-features = false }
+ink_lang = { version = "~3.4.0", default-features = false }
+ink_prelude = { version = "~3.4.0", default-features = false }
+
+scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
+scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
+
+# use forked 2.3.0 version of openbrush until asset_id is revolved
+openbrush = { git = "https://github.com/PierreOssun/openbrush-contracts", default-features = false, features = ["psp22"] }
```
Comment (Maar-io) [inline contracts/psp22_pallet_wrapper/Cargo.toml:19]: dependency on local repo. Remove before merge

File: contracts/psp22_pallet_wrapper/lib.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            self._mint_to(caller, amount)
+        }
+
+        #[ink(message)]
+        pub fn withdraw(&mut self, amount: Balance) -> Result<(), PSP22Error> {
+            let caller = self.env().caller();
+            self._burn_from(caller, amount)?;
+            self.pallet_assets
+                .transfer(Origin::Address, self.asset_id, caller, amount)
+                .map_err(|_| PSP22Error::Custom("transfer failed".as_bytes().to_vec()))
```
Comment (Maar-io) [inline contracts/psp22_pallet_wrapper/lib.rs:97]:
```suggestion
                .map_err(|_| PSP22Error::Custom("transfer failed".as_bytes().to_vec()))
```

Comment (Maar-io) [inline contracts/psp22_pallet_wrapper/lib.rs:97]: you can use `into()` for String conversion

`.map_err(|_|` PSP22Error::Custom("transfer failed".into()))

File: contracts/psp22_pallet_wrapper/lib.rs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            self.asset_id
+        }
+
+        #[ink(message)]
+        pub fn deposit(&mut self, amount: Balance) -> Result<(), PSP22Error> {
+            let caller = self.env().caller();
+            let contract = self.env().account_id();
+            self.pallet_assets
+                .transfer(Origin::Caller, self.asset_id, contract, amount)
+                .map_err(|_| PSP22Error::Custom("transfer failed".as_bytes().to_vec()))?;
```
Comment (Maar-io) [inline contracts/psp22_pallet_wrapper/lib.rs:87]: you can use `into() `for String conversion

`                .map_err(|_| PSP22Error::Custom("transfer failed".into()))?;
`

**Review by Maar-io — APPROVED** [review]
LGTM

**Pull request merged**